### ISSUE TITLE
Made metadata fields that are arrays display as unordered lists

### ISSUE THREAD

**Issue by erinchambers** [CDRH/orchid#336]
...or we could use `<dt>`/`<dd>` instead.

Right now, these lists are plain text separated by `<span>`s and `|` characters, which means that 
a) screen readers don't inform users that there are multiple values (e.g. multiple authors)
b) screen readers read the list in a haphazard way because `span` and `|` are not announced and don't affect the cadence when the text is read aloud. Commas, however, _do_ affect cadence; they are interpreted by screen readers as pauses. Therefore, when it is a list of names that are "Last Name, First Name," screen readers pause between person 1's last name and first name, not between person 1 and person 2.

**Comment by erinchambers** [comment]
I tried messing around with this in `display_helper.erb` but didn't get the result I expected. It'd be awesome if someone could help me out!

I wonder if this is actually accomplished through XSLT for some projects, too. If we make this improvement in Orchid, we might want to be on the lookout for those.

**Comment by erinchambers** [comment]
Decided at 2025-10-29 dev team checkin to do this, and just do it across the board for all metadata fields; no need to distinguish between which fields are arrays and which ones are single values.

In Teams on 2025-10-31, decided
- not to create a partial to handle the extra lines of html, since there are only two (`<ul>` and `</ul>`); just set it up within the helper
- to leave the `separator` code in, even though we're currently making it blank (`separator: ""`) and changing the `join` to `.join("")`, because we might want to use it in the future.

Fixed on `messing_with_helper` branch and merged into accessibility branches.
